src/generic/stage1/utf8_lookup4_algorithm.h

```diff
@@ -139,7 +139,8 @@ using namespace simd;
       this->error |= check_multibyte_lengths(input, prev_input, sc);
     }
 
-    // The only problem that can happen at EOF is that a multibyte character is too short.
+    // The only problem that can happen at EOF is that a multibyte character is too short
+    // or a byte value too large.
```

Comment:
Byte value too large? Like an ASCII character at the end? Curious what this case is, I can't visualize it.

Reply:
`check_special_cases` only checks for bytes too large in the first of two bytes.

Reply:
@jkeiser I have added an explanation.

Reply:
Sure, but byte values that are too large can't happen at the end of the file, right?

I guess this is implicitly validating that the final byte is not too long, but by that metric it's also checking whether a 2-, 3- or 4-byte value at the end of the file is not too long :)

Reply:
Let me be clear. The lookup4 algorithm is sound. We were just applying it wrongly by not executing the oef check. It is probably my fault too.

> I guess this is implicitly validating that the final byte is not too long, but by that metric it's also checking whether a 2-, 3- or 4-byte value at the end of the file is not too long :)

That's correct. It does all these checks at once.

> Sure, but byte values that are too large can't happen at the end of the file, right?

If you omit the EOF check, you could have something like this...

"0x20 0x20 0x20 0x20..... 0x20 0x20 0xFF"

Now, that's not a multibyte character that is too short.

that would pass.

Reply:
Oh! You are talking about invalid leads. Got it. Makes sense.

I sometimes think of 0xFF as an 8-byte lead; I find it can make it easier to think about things. But it's neither here nor there. 👍